### ISSUE TITLE
Test using VS Code remote support / dev containers

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#1677]
https://code.visualstudio.com/blogs/2019/05/02/remote-development
https://code.visualstudio.com/api/advanced-topics/remote-extensions
https://code.visualstudio.com/docs/remote/containers

"Dev Containers" are apparently the most restrictive, so we should start there to find most issues. We'll definitely need to fix the `openInBrowser` calls to call `openExternal` (in a way we can still stub for testing).

Things likely to be broken in this setup:

- [ ] calls to `openInBrowser`
- [ ] having webdev launch browsers
- [ ] having DevTools launch browsers
- [ ] launching emulators
- [ ] running 'Open in Xcode' / 'Open in Android Studio'
- [ ] connecting to devices on the local machine

Mostly blocked by:

- https://github.com/microsoft/vscode/issues/74258
- https://github.com/microsoft/vscode/issues/74188

**Comment by minkir014** [comment]
Do you mean you want to support remote unit testing for dart and flutter?

**Comment by DanTup** [comment]
I'm not sure exactly what you have in mind, but this issue is just to make sure the extension works if people are using it with the "remote development" features linked to above (eg. VS Code is connected to a remote workspace with the Dart VM and project on the "server"). There's at least one thing I don't think will work (spawning browsers) because we don't use the new API for it.

Whether this will support what you're after here, I'm not sure, because I'm not sure what it means to support remote unit testing. Presumably with the right "server" you could do whatever you want - but this issue is not related to providing any specific servers, only to ensure the extension can run in the "backend only" environment VS Code uses for these features.

(If there's something you have in mind that this doesn't cover, feel free to open new issue(s)!)

**Comment by minkir014** [comment]
You mean flutter web apps with remote technology may have a problem.

**Comment by minkir014** [comment]
Am I right? Or you're opening an issue only for need.

**Comment by minkir014** [comment]
@DanTup Are you here?

**Comment by DanTup** [comment]
> You mean flutter web apps with remote technology may have a problem.

The specific example I gave above was for when VS Code launches a browser (for example launching DevTools). However you've just reminded me that webdev also launches browser and that don't work here (they'd launch in the container) so we'd need some solution to that too (I don't know what that will be, and the priority of it may depend on how common this setup is).

If there's anything specific you're after or if you're seeing issues, do feel free to open new issues. More issues with possible duplicates is better than things being lost mixed in with other issues :-)

**Comment by DanTup** [comment]
The stuff that can work all now works, and the things that don't are disabled (https://github.com/Dart-Code/Dart-Code/issues/1738) with the exception of DevTools (I've opened https://github.com/Dart-Code/Dart-Code/issues/1866, but it's blocked on VS Code).

Support for things like Flutter web will be added if the underlying tools get support (currently it's not possible to run Flutter web on a remote machine because it spawns Chrome to use the debug protocol, for example).
